File: benchmarks/c/arithmetic_loops.c
#include <stdint.h>
#include <stdio.h>

static int64_t mix(int64_t value, int64_t index) {
    value = value + index * 17;
    value = value ^ (index << 2);
    value = value - (index % 11);
    return value & 1048575;
}

int main(void) {
    int64_t result = 1;
    for (int64_t i = 0; i < 200000; ++i) {
        result = mix(result, i);
    }
    printf("%lld\n", (long long) result);
    return 0;
}

File: benchmarks/python/arithmetic_loops.py
MASK64 = (1 << 64) - 1
SIGN64 = 1 << 63


def i64(value: int) -> int:
    value &= MASK64
    if value >= SIGN64:
        return value - (1 << 64)
    return value


def mix(value: int, index: int) -> int:
    value = i64(value + i64(index * 17))
    value = i64(value ^ i64(index << 2))
    value = i64(value - (index % 11))
    return i64(value & 1048575)


def main() -> None:
    result = 1
    for i in range(200000):
        result = mix(result, i)
    print(result)


if __name__ == "__main__":
    main()
